classes/Mail.php

```diff
@@ -212,29 +210,19 @@ public static function Send($id_lang, $template, $subject, $template_vars, $to,
                     Tools::dieOrLog(Tools::displayError('Error: invalid SMTP server or SMTP port'), $die);
                     return false;
                 }
-                $connection = new Swift_Connection_SMTP(
-                    $configuration['PS_MAIL_SERVER'],
-                    $configuration['PS_MAIL_SMTP_PORT'],
-                    $configuration['PS_MAIL_SMTP_ENCRYPTION'] == 'ssl' ? Swift_Connection_SMTP::ENC_SSL : (($configuration['PS_MAIL_SMTP_ENCRYPTION'] == 'tls' ? Swift_Connection_SMTP::ENC_TLS : Swift_Connection_SMTP::ENC_OFF))
-                );
-                $connection->setTimeout(4);
-                if (!$connection) {
-                    return false;
-                }
-                if (!empty($configuration['PS_MAIL_USER'])) {
-                    $connection->setUsername($configuration['PS_MAIL_USER']);
-                }
-                if (!empty($configuration['PS_MAIL_PASSWD'])) {
-                    $connection->setPassword($configuration['PS_MAIL_PASSWD']);
-                }
+
+                $connection = Swift_SmtpTransport::newInstance($configuration['PS_MAIL_SERVER'], $configuration['PS_MAIL_SMTP_PORT'], $configuration['PS_MAIL_SMTP_ENCRYPTION'])
+                    ->setUsername($configuration['PS_MAIL_USER'])
+                    ->setPassword($configuration['PS_MAIL_PASSWD']);
+
             } else {
-                $connection = new Swift_Connection_NativeMail();
+                $connection = Swift_SmtpTransport::newInstance('localhost', 25);
```

Comment:
hmm I need to think a little bit about it, *but* as I wanted to update SwitfMailer, I'm glad to see you did.

My only other concern is the branch of PrestaShop concerned by this improvement: we are - the core team - actualy focused on the next major release and this contribution seems to big to reach the next 1.6 (1.6.1.5, right ?).

ping @Shudrum : how about the "acceptability" of this contribution against 1.6.1.x ?